include/ttmlir/Dialect/TTNN/Analysis/ShardSolver.h

```diff
@@ -310,6 +313,9 @@ class ShardSolver {
 
   llvm::DenseMap<Operation *, TTNNLayoutAttr> selectedOpLayout;
   std::unordered_set<Edge> memReconfigEdges;
+  std::function<bool(mlir::Operation *, TTNNLayoutAttr const &,
+                     mlir::Operation *, TTNNLayoutAttr const &)>
+      customCheckShardCompatible;
```

Comment:
Are we going to fetch the second operand within this function or change the function signature to support this in the future?

Reply:
Signature does not have to change, you can invoke getOperands from consumer operation.